src/v/kafka/server/connection_context.cc

```diff
@@ -302,6 +302,13 @@ connection_context::dispatch_method_once(request_header hdr, size_t size) {
                             });
                       })
                       .handle_exception([self](std::exception_ptr e) {
+                          // ssx::spawn_with_gate already caught shutdown-like
+                          // exceptions, so we should only be taking this
+                          // path for real errors.  That also means
+                          // that on shutdown we don't bother to call
+                          // shutdown_input on the connection, so rely
+                          // on any future reader to check the abort
+                          // source before considering reading the connection.
```

Comment:
> Would be good to get a second pair of eyes on this: I don't think the shutdown_input call is necessary to progress shutdown of the overall service, right?

I think this should be ok. What was nice about this before is that if any exception occurred in background processing (which covers all kinds of kafka api handlers) then the connection would be shutdown so new handling would be terminated quickly.

in this case it looks like even if something like a broken semaphore bubbled up, requests on the connection might continue to be served. what if those requests were handled in the background here and continually hit the same exceptional condition but we ignored it and didn't shutdown the connection?